extensions/amp-install-serviceworker/0.1/amp-install-serviceworker.js

```diff
@@ -290,50 +290,110 @@ class UrlRewriter_ {
  * Installs the service worker at src via direct service worker installation.
  * @param {!Window} win
  * @param {string} src
+ * @param {Element} element
  * @return {!Promise<!ServiceWorkerRegistration|undefined>}
  */
-function install(win, src) {
+function install(win, src, element) {
   return win.navigator.serviceWorker.register(src).then(function(registration) {
     if (getMode().development) {
       user().info(TAG, 'ServiceWorker registration successful with scope: ',
           registration.scope);
     }
-    sendAmpScriptToSwOnFirstVisit(registration, win);
+    // Check if there is a new service worker installing.
+    const installingSw = registration.installing;
+    if (installingSw) {
+      // if not already active, wait till it becomes active
+      installingSw.addEventListener('statechange', evt => {
```

Comment:
There appears to be an `activate` event on the registration itself.

Reply:
cant find one: https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerRegistration#Properties

Reply:
Cmd+f "activate". 😛

Reply:
ahhh gotcha... i was looking for a method `onactivate` instead of the event name... gotcha

Reply:
I am unable to trigger this event by any means, will stick to current code if cannot find a reliable means to trigger this

Reply:
SGTM